ui/app/store/actions.js

```diff
@@ -1382,6 +1382,11 @@ export function addToken(
   dontShowLoadingIndicator,
 ) {
   return (dispatch) => {
+    if (!address) {
```

Comment:
This change alone should prevent adding tokens without an address.

If there's uncertainty about where in the UI this was made possible (i.e. whether it was in swaps, or the confirm suggested token, or both) we could ship just this change and the migration. Then we'd be preventing invalid state, and we'd get nice helpful Sentry errors for any remaining UI bugs.